lib/services/suite/suite.go

```diff
@@ -1011,6 +1013,124 @@ func (s *ServicesTestSuite) ClusterConfig(c *check.C, opts ...SuiteOption) {
 	fixtures.DeepCompare(c, clusterName, gotName)
 }
 
+// SemaphoreContention checks that a large number of concurrent acquisitions
+// all succeed if MaxLeases is sufficiently high.  Note that we do not test
+// the same property holds for releasing semaphores; release operations are
+// best-effort and allowed to fail.
+func (s *ServicesTestSuite) SemaphoreContention(c *check.C) {
+	const locks int64 = 20
```

Comment:
Since this is a stress test of sorts, can you bump it to ~1000?

Reply:
Bumped to 5 separate iterations of 50 each.  The auth server starts returning "too much contention" errors at around 100, but that is by design... we don't want bad code to cause the auth server sit for long periods of time on hundreds of requests.  Contention only happens on a per-semaphore basis, so a user would need to start >100 separate SSH connections within a few seconds to hit this with `sessctl`.